langchain4j-core/src/main/java/dev/langchain4j/agent/tool/ToolSpecifications.java

```diff
@@ -44,6 +67,23 @@ public static List<ToolSpecification> toolSpecificationsFrom(Object objectWithTo
         return toolSpecificationsFrom(objectWithTools.getClass());
     }
 
+    /**
+     * Validates all the {@link ToolSpecification}s. The validation checks for duplicate method names.
+     * Throws {@link IllegalArgumentException} if validation fails
+     *
+     * @param toolSpecifications list of ToolSpecification to be validated.
+     */
+    public static void validateSpecifications(List<ToolSpecification> toolSpecifications) throws IllegalArgumentException {
+
+        // Checks for duplicates methods
+        Set<String> names = new HashSet<>();
+        for (ToolSpecification toolSpecification : toolSpecifications) {
+            if (!names.add(toolSpecification.name())) {
+                throw new IllegalArgumentException(format("Methods in Tools must be unique. The method '%s' appears several times", toolSpecification.name()));
```

Comment:
```suggestion
                throw new IllegalArgumentException(format("Tool names must be unique. The tool '%s' appears several times", toolSpecification.name()));
```
Method names are used as tool names by default, but they can be overriden:
```java
        @Tool(name = "a")
        public int duplicateMethod(String typeString) {
            return 42;
        }

        @Tool(name = "b")
        public int duplicateMethod(int typeInt) {
            return 42;
        }
```
So I would suggest changing the error message to better reflect the validation logic.